ui/src/flux/components/flux_functions_toolbar/SearchBar.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {PureComponent, ChangeEvent} from 'react'
+import _ from 'lodash'
+
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+interface Props {
+  onSearch: (s: string) => void
+}
+interface State {
+  searchTerm: string
+}
+
+@ErrorHandling
+class SearchBar extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      searchTerm: '',
+    }
+  }
+
+  public componentWillMount() {
+    const waitPeriod = 100
+    this.handleSearch = _.debounce(this.handleSearch, waitPeriod)
+  }
+
+  public render() {
+    return (
+      <div className="flux-functions-toolbar--search">
+        <div className="search-widget">
+          <input
```

Comment:
Use the ui kit input, it has loading and error states, also has icons

Reply:
replaced